great_expectations/rule_based_profiler/config/base.py

```diff
@@ -506,16 +506,16 @@ def __init__(
         BaseYamlConfig.__init__(self, commented_map=commented_map)
 
     @classmethod
-    def from_commented_map(cls, commented_map: CommentedMap):  # type: ignore[no-untyped-def]
+    def from_commented_map(cls, commented_map: CommentedMap):  # type: ignore[override] # super type accepts Dict
         """Override parent implementation to pop unnecessary attrs from config.
 
         Please see parent BaseYamlConfig for more details.
         """
         try:
-            schema_instance: Any = cls._get_schema_instance()
+            schema_instance = cls._get_schema_instance()
             config: Union[dict, BaseYamlConfig] = schema_instance.load(commented_map)
-            config.pop("class_name", None)
-            config.pop("module_name", None)
+            config.pop("class_name", None)  # type: ignore[union-attr] # BaseYamlConfig has no `.pop()`
```

Comment:
@Kilo59 Are we certain about that?  Given that `BaseYamlConfig` inherits dictionary behavior and `__delitem__()` is implemented, should `pop` not be automatically "understood"?  If not, then it is probably because it is not explicitly implemented, which might be confusing the type checker.  Thanks.

Reply:
https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

`DictDot` doesn't inherit from `dict` or any other `MutableSequence` that contains a `pop()`.

https://github.com/great-expectations/great_expectations/blob/c659da680fb20233a965231b1728a9f15fc32ff1/great_expectations/types/__init__.py#L24